device/api/umd/device/cluster.h

```diff
@@ -648,13 +634,12 @@ class Cluster : public tt_device {
     virtual bool using_harvested_soc_descriptors();
     virtual std::unordered_map<chip_id_t, uint32_t> get_harvesting_masks_for_soc_descriptors();
     virtual void translate_to_noc_table_coords(chip_id_t device_id, std::size_t& r, std::size_t& c);
-    virtual int get_number_of_chips_in_cluster();
-    virtual std::unordered_set<chip_id_t> get_all_chips_in_cluster();
     virtual tt_ClusterDescriptor* get_cluster_description();
     static int detect_number_of_chips();
     static std::vector<chip_id_t> detect_available_device_ids();
     virtual std::set<chip_id_t> get_target_mmio_device_ids();
     virtual std::set<chip_id_t> get_target_remote_device_ids();
+    virtual std::set<chip_id_t> get_target_device_ids();
```

Comment:
nit: maybe rename to `get_chips_ids`

Reply:
we are using name `device` throughout the whole class to actually refer to a chip. I don't want to change this now, we might change all of these at some point.

And `target` is there because this refers to only the chips chosen in the constructor to be used. As opposed to `detect_available_device_ids` which returns all chips available on system. I also don't want to change other _target_ functions, because that would lead to an API change which I'd have to address in tt_metal, and this PR doesn't have an API change for now. Leaving others with "target" in name would then be another inconsistency.

We can consider changing `get_target_mmio_device_ids` `get_target_remote_device_ids` `get_target_device_ids` and `detect_available_device_ids`, but I'd leave it out of this PR for now.